Importer/Log/Log.php

```diff
@@ -0,0 +1,128 @@
+<?php
+namespace Netdudes\ImporterBundle\Importer\Log;
+
+class Log
+{
+    /**
+     * @var string[]
+     */
+    protected $configurationErrors = [];
+
+    /**
+     * @var string[]
+     */
+    protected $dataErrors = [];
+
+    /**
+     * @var string[]
+     */
+    protected $criticalErrors = [];
+
+    /**
+     * @var string
+     */
+    protected $csvLines;
+
+    /**
+     * @var array
+     */
+    protected $importedEntities;
+
+    /**
+     * @param string $errorMessage
+     */
+    public function addConfigurationError($errorMessage)
+    {
+        $this->configurationErrors[] = $errorMessage;
+    }
+
+    /**
+     * @param int    $index
+     * @param string $errorMessage
+     */
+    public function addDataError($index, $errorMessage)
+    {
+        $this->dataErrors[$index] = [
+            'index_data' => $this->getIndexData($index),
+            'error_message' => $errorMessage,
+        ];
+    }
+
+    /**
+     * @param string $message
+     */
+    public function addCriticalError($message)
+    {
+        $this->criticalErrors[] = $message;
```

Comment:
On 'u2', when we add a critical error we also want this to be logged. 
I thought it's better not to do so here.
This means that in 'u2' I have to override this function to:
- firstly log the message
- and then call the parent

What do you think?